src/NServiceBus.RabbitMQ/Receiving/MessageConverter.cs

```diff
@@ -33,36 +33,44 @@ public string RetrieveMessageId(BasicDeliverEventArgs message, Dictionary<string
         public Dictionary<string, string> RetrieveHeaders(BasicDeliverEventArgs message)
         {
             var properties = message.BasicProperties;
+            var headers = properties.Headers;
```

Comment:
Does the `Headers` property have side effects, or does is just return the value of a field? If the latter (and what the semantics of a property getter convey) then it should be fine to call it more than once.

@SeanFeldman I don't have a problem with the `properties` variable, only `headers` and `messageHeaders`.

If we were to rename variables, then I'd probably rename `headers` to `objectHeaders` and `messageHeaders` to `stringHeaders`, since `object` vs `string` is the only difference between them. Or possibly `originalHeaders` and `deserializedHeaders`.

However, this somewhat of a bike shed so let's not spend too much time on it. Unless the task force feels that one of my suggestions (drop the `headers` variable or one of the two renames) is an obvious improvement, let's just leave as-is.